cmd/main.go

```diff
@@ -555,3 +566,82 @@ func createControllerManager() (manager.Manager, error) {
 	}
 	return nil, nil
 }
+
+func newAssistedServiceRoute(service *corev1.Service) *routev1.Route {
+	return &routev1.Route{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      service.Name,
+			Namespace: service.Namespace,
+			Labels:    map[string]string{"app": service.Name},
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion: "v1",
+					Kind:       "Service",
+					Name:       service.Name,
+					UID:        service.UID,
+				},
+			}},
+		Spec: routev1.RouteSpec{
+			Port: &routev1.RoutePort{
+				TargetPort: intstr.FromString(service.Name)},
+			To: routev1.RouteTargetReference{
+				Kind: "Service",
+				Name: service.Name},
+			TLS: &routev1.TLSConfig{
+				Termination:                   routev1.TLSTerminationEdge,
+				InsecureEdgeTerminationPolicy: routev1.InsecureEdgeTerminationPolicyAllow}},
+		Status: routev1.RouteStatus{},
+	}
+}
+
+func createRouteAndUpdateServiceBaseURL(log logrus.FieldLogger, failOnError func(err error, msg string, args ...interface{})) {
+	log.Info("Creating route for service")
+
+	failOnError(routev1.AddToScheme(scheme.Scheme),
+		"Failed to add routev1 to scheme")
+
+	client, err := client.New(config.GetConfigOrDie(), client.Options{Scheme: scheme.Scheme})
+	failOnError(err, "failed to create controller-runtime client")
+
+	ctx := context.Background()
+
+	serviceName := "assisted-service"
+	namespace := "assisted-installer"
```

Comment:
Do we always deploy to this namespace or is it something optional?

Reply:
excellent point. This should probably come using the downward API.

Reply:
> Just making sure that we have permissions to edit Hive cluster deployment.

Yes, they are included in the CSV generated in the bundle. More specifically, the permissions in internal/controller/config/rbac/role.yaml are included in the CSV.

Reply:
The code has been replaced with the initContainers which are namespace aware.